Sources/KSCrashRecording/Monitors/KSCrashMonitor_User.c

```diff
@@ -87,6 +88,7 @@ void kscm_reportUserException(const char *name, const char *reason, const char *
         if (terminateProgram) {
             abort();
         }
+        kscm_clearAsyncSafetyState();
```

Comment:
What if a user reports more then 2 exceptions with `logAllThreads`? Should we regenerate IDs here?

Reply:
I'm putting in a transitional API for this. This whole issue will go away under the new design.